test/infrastructure/docker/api/v1beta1/dockermachine_types.go

```diff
@@ -30,6 +30,13 @@ const (
 
 // DockerMachineSpec defines the desired state of DockerMachine.
 type DockerMachineSpec struct {
+	// InstanceName indicates the name of the Docker container associated with this DockerMachine.
+	// Since the provider ID is not set until after the container is online, this field is used to
+	// maintain the association. If it is not populated, the name of the container will be the same
+	// as the name of the owner Machine.
+	// +optional
+	InstanceName string `json:"instanceName,omitempty"`
```

Comment:
This seems that we are lacking in the API something that is an internal implementation detail, I need to think more about this 🤔

Reply:
Any thoughts? I think the main thing is that we want to have some way in the spec to associate the DockerMachine with the docker container in this line:
```
externalMachine, err := docker.NewMachine(ctx, cluster, dockerMachine.Spec.InstanceName, nil)
```
Before, we were using the machine name but for MachinePool Machines, those machines don't exist yet.